gems/pending/spec/postgres_ha_admin/failover_monitor_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'postgres_ha_admin/failover_monitor'
+require 'util/postgres_admin'
+
+describe PostgresHaAdmin::FailoverMonitor do
+  let(:db_yml) do
+    yml = double('DatabaseYml')
+    allow(yml).to receive(:pg_params_from_database_yml).and_return(:host => 'host.example.com', :user => 'root')
+    yml
+  end
+  let(:failover_db) { double('FailoverDatabases') }
+  let(:connection) do
+    conn = double("PGConnection")
+    allow(conn).to receive(:finish)
+    conn
+  end
+
+  let(:failover_monitor) do
+    expect(PostgresHaAdmin::DatabaseYml).to receive(:new).and_return(db_yml)
+    expect(PostgresHaAdmin::FailoverDatabases).to receive(:new).and_return(failover_db)
+    failover_instance = described_class.new('', '', @logger_file.path, 'test')
+    logger = double('Logger')
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:error)
+    failover_instance.instance_variable_set("@logger", logger)
+    failover_instance
+  end
+
+  let(:linux_admin) do
+    linux_adm = double('LinuxAdmin')
+    allow(LinuxAdmin::Service).to receive(:new).and_return(linux_adm)
+    linux_adm
+  end
+  
+  before do
+    @logger_file = Tempfile.new('ha_admin.log')
+  end
+
+  after do
+    @logger_file.close(true)
+  end
+
+  describe "#monitor" do
+    context "primary database is accessable" do
+      before do
+        allow(PG::Connection).to receive(:open).and_return(connection)
+      end
+
+      it "updates 'failover_databases.yml'" do
+        expect(failover_db).to receive(:update_failover_yml)
+        failover_monitor.monitor
+      end
+
+      it "does not stop evm server and does not execute failover" do
+        expect(failover_db).to receive(:update_failover_yml)
+        expect(linux_admin).not_to receive(:stop)
+        expect(linux_admin).not_to receive(:restart)
+        expect(failover_monitor).not_to receive(:execute_failover)
+
+        failover_monitor.monitor
+      end
+    end
+
+    context "primary database is not accessable" do
+
+      before do
+        allow(PG::Connection).to receive(:open).and_return(nil, connection, connection)
+      end
+
+      describe "#host_is_repmgr_primary?" do
```

Comment:
Makes sense to me, then we can make `query_repmgr` a private method again right?